File: Backend/MedicineDelivery/MedicineDelivery.Domain/Entities/Payment.cs
using MedicineDelivery.Domain.Enums;

namespace MedicineDelivery.Domain.Entities
{
    public class Payment
    {
        public int PaymentId { get; set; }
        public int OrderId { get; set; }
        public string PaymentMode { get; set; } = string.Empty;
        public string TransactionId { get; set; } = string.Empty;
        public decimal Amount { get; set; }
        public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Pending;
        public DateTime PaidOn { get; set; } = DateTime.UtcNow;

        public Order? Order { get; set; }
    }
}

File: WebApp/src/models/Payment.ts
export interface Payment {
  paymentId: string;
  orderId: string;
  paymentMode?: string;
  transactionId?: string;
  amount: number;
  paymentStatus: number; // 0 = Pending, 1 = Success, 2 = Failed
  paidOn?: string;
}
